ibtk/include/ibtk/HierarchyMathOps.h

```diff
@@ -224,10 +224,10 @@ class HierarchyMathOps : public SAMRAI::tbox::DescribedClass
      * \see resetLevels
      */
     void curl(int dst_idx,
-              SAMRAI::tbox::Pointer<SAMRAI::pdat::CellVariable<NDIM, double> > dst_var,
+              const SAMRAI::pdat::CellVariable<NDIM, double>* dst_var,
```

Comment:
This is a good opportunity to switch these to references. Like I noted yesterday, one problem with this interface is that

```cpp
curl(0, 0, 0, 0, 0, 0);
```
compiles due to implicit conversions. We should leverage the type system here.

Is there a case in which we would know the type of variable but not have the `Pointer<Variable>` on-hand? I don't think I have ever seen that in IBAMR.